<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <!-- quad name -->
    <!-- <arg name="ns"  default="$(env HOSTNAME)"/> -->
    <!-- GAZEBO: -->
    <arg name="ns"  default="drone2"/>
    <!-- topic names -->
    <arg name="targetPose_topic" default="command/pose"/>
    <arg name="targetTwist_topic" default="command/twist"/>
    <arg name="currentPose_topic" default="mavros/vision_pose/pose"/>
    <arg name="carpService_service" default="carpService"/>
    <!-- timer rates -->
    <arg name="setpointFreq" default="100.0"/>

    <arg name="kp" default="1.0"/>
    <arg name="ki" default="0.0002"/>
    <arg name="kd" default="0.0"/>
    <arg name="integralLimit" default="1.0"/>

    <group ns="$(arg ns)">
        <node pkg="bridge_px4" type="controller" name="controller" output="screen">
            <param name="targetPose_topic" value="$(arg targetPose_topic)"/>
            <param name="targetTwist_topic" value="$(arg targetTwist_topic)"/>
            <param name="currentPose_topic" value="$(arg currentPose_topic)"/>
            <param name="setpointFreq" value="$(arg setpointFreq)"/>
            <param name="kp" value="$(arg kp)"/>
            <param name="ki" value="$(arg ki)"/>
            <param name="kd" value="$(arg kd)"/>
            <param name="integralLimit" value="$(arg integralLimit)"/>
        </node>

    </group>
</launch>
